src/reference/asciidoc/mongodb.adoc

```diff
@@ -406,4 +408,17 @@ public class MongoDbJavaApplication {
     }
 
 }
+----
+
+Alternatively to the query and query-expression properties, you can specify other database operations through
+the collectionCallback property. The following example specifies a count operation:
+
+[source, java]
+----
+private MongoDbOutboundGatewaySpec collectionCallbackOutboundGateway() {
+        return MongoDb.outboundGateway(this.mongoDbFactory, this.mongoConverter)
+                .collectionCallback(MongoCollection::count)
+                .collectionName("foo")
+                .entityClass(Person.class);
```

Comment:
Although it doesn't hurt, and is just ignored, the `entityClass` doesn't make sense when `collectionCallback` is in use.
So, just remove it from here since it is a sample for end-users, who may make a wrong decision about the premise of the sample.